flink-runtime/src/test/java/org/apache/flink/runtime/operators/AbstractOuterJoinTaskExternalITCase.java

```diff
@@ -0,0 +1,121 @@
+import org.apache.flink.runtime.operators.testutils.BinaryOperatorTestBase;
+import org.apache.flink.runtime.operators.testutils.UniformIntTupleGenerator;
+import org.apache.flink.util.Collector;
+import org.junit.Assert;
+import org.junit.Test;
+
+public abstract class AbstractOuterJoinTaskExternalITCase extends BinaryOperatorTestBase<FlatJoinFunction<Tuple2<Integer, Integer>,
+		Tuple2<Integer, Integer>, Tuple2<Integer, Integer>>, Tuple2<Integer, Integer>, Tuple2<Integer, Integer>> {
+
+	private static final long HASH_MEM = 4*1024*1024;
+
+	private static final long SORT_MEM = 3*1024*1024;
+
+	private static final long BNLJN_MEM = 10 * PAGE_SIZE;
+
+	private final double bnljn_frac;
+
+	@SuppressWarnings("unchecked")
+	private final TypeComparator<Tuple2<Integer, Integer>> comparator1 = new TupleComparator<>(
+			new int[]{0},
+			new TypeComparator<?>[] { new IntComparator(true) },
+			new TypeSerializer<?>[] { IntSerializer.INSTANCE });
+
+	@SuppressWarnings("unchecked")
+	private final TypeComparator<Tuple2<Integer, Integer>> comparator2 = new TupleComparator<>(
+			new int[]{0},
+			new TypeComparator<?>[] { new IntComparator(true) },
+			new TypeSerializer<?>[] { IntSerializer.INSTANCE });
+
+	private final TypeSerializer<Tuple2<Integer, Integer>> serializer = new TupleSerializer<>(
+			(Class<Tuple2<Integer, Integer>>) (Class<?>) Tuple2.class,
+			new TypeSerializer<?>[] { IntSerializer.INSTANCE, IntSerializer.INSTANCE });
+
+	private final CountingOutputCollector output = new CountingOutputCollector();
+
+	private final DriverStrategy driverStrategy;
+
+	public AbstractOuterJoinTaskExternalITCase(ExecutionConfig config, DriverStrategy driverStrategy) {
+		super(config, HASH_MEM, 2, SORT_MEM);
+		bnljn_frac = (double)BNLJN_MEM/this.getMemoryManager().getMemorySize();
+		this.driverStrategy = driverStrategy;
+	}
+
+	@Test
+	public void testExternalSort1OuterJoinTask() {
+		final int keyCnt1 = 16384*4;
+		final int valCnt1 = 2;
+		
+		final int keyCnt2 = 8192;
+		final int valCnt2 = 4*2;
+		
+		final int expCnt = calculateExpectedCount(keyCnt1, valCnt1, keyCnt2, valCnt2);
+
+		setOutput(this.output);
+		addDriverComparator(this.comparator1);
+		addDriverComparator(this.comparator2);
+		getTaskConfig().setDriverPairComparator(new RuntimePairComparatorFactory());
+		getTaskConfig().setDriverStrategy(driverStrategy);
+		getTaskConfig().setRelativeMemoryDriver(bnljn_frac);
+		setNumFileHandlesForSort(4);
+
+		final AbstractOuterJoinDriver<Tuple2<Integer, Integer>, Tuple2<Integer, Integer>, Tuple2<Integer, Integer>> testTask =
+				getOuterJoinDriver();
+		
+		try {
+			addInputSorted(new UniformIntTupleGenerator(keyCnt1, valCnt1, false), serializer, this.comparator1.duplicate());
+			addInputSorted(new UniformIntTupleGenerator(keyCnt2, valCnt2, false), serializer, this.comparator2.duplicate());
+			testDriver(testTask, MockJoinStub.class);
+		} catch (Exception e) {
+			e.printStackTrace();
+			Assert.fail("The test caused an exception.");
```

Comment:
Catching an exception, printing its stack trace and then failing the test with an `AssertionError` is not a good pattern, because that way the relevant stack trace will be hidden somewhere in the stdout output. Why not simply letting the exception bubble up. This will also fail the test case.